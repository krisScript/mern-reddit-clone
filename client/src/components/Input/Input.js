import React, { useState, useMemo } from 'react';
const errorFunc = (validationErrorParams, name) => {
  let error = false;
  if (validationErrorParams && validationErrorParams.length > 0) {
    if (validationErrorParams.includes(name)) {
      error = true;
    }
  }
  return error;
};
const Input = ({ name,
  setHook,
  placeholder,
  value,
  type,
  validationErrorParams,
  textArea}) => {
  
  const error = useMemo(() => errorFunc(validationErrorParams, name), [
    validationErrorParams,
    name
  ]);
  let component = (
    <input
      className={`input${error ? ' input--error' : ''}`}
      onChange={e => setHook(e.target.value)}
      value={value}
      type={type}
      placeholder={placeholder}
      name={name}
      // required
    />
  );
  if (textArea) {
    component = (
      <textarea
        className={`input${error ? ' input--error' : ''}`}
        onChange={e => setHook(e.target.value)}
        type={type}
        placeholder={placeholder}
        name={name}
        rows={7}
        value={value}
        // required
      />
    );
  }
  return component;
};
export default Input;
