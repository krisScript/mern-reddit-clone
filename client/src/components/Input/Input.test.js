import React from 'react'

import {render,cleanup} from 'react-testing-library'
import { BrowserRouter } from 'react-router-dom';
import Input from './Input'

describe('<Input />',() => {
    describe('without error',() => {
    const setHook = jest.fn()
    const value= 'test'
    const type= 'text'
    const placeholder= 'Test Placeholder'
    const name= 'test'
    let errorArr = []
    const { container, rerender,queryByPlaceholderText,queryByValue ,getByPlaceholderText } = render (<Input setHook={setHook} value={value} type={type} placeholder={placeholder} errorArr={errorArr} name={name} />)
    beforeEach(() => {    
        
    })
    it('should have placeholder text equal to placeholder',() => {
        expect(queryByPlaceholderText(placeholder)).toBeTruthy()
    })
    it('should have value text equal to value',() => {
        expect(queryByValue(value)).toBeTruthy()
    })
    it('should have input class',() => {
        expect(getByPlaceholderText(placeholder).className).toMatch('input')
    })
    it('should have name equal to name',() => {
        expect(getByPlaceholderText(placeholder).name).toMatch(name)
    })

    })
    //////Currently dosn't work  /////

    describe('with error',() => {
        const setHook = jest.fn()
    const value= 'test'
    const type= 'text'
    const placeholder= 'Test Placeholder'
    const name= 'test'
        const errorArr = ['test']
        const { getByPlaceholderText } = render (<Input setHook={setHook} value={value} type={type} placeholder={placeholder} errorArr={errorArr} name={name}/>)
        it('should have input class',() => {
            console.log(getByPlaceholderText(placeholder))
            expect(getByPlaceholderText(placeholder).className).toMatch('input')
        })
    })
})
