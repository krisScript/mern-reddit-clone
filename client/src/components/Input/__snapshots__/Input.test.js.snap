// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> snapshot 1`] = `
<div>
  <input
    class="input"
    name="test"
    placeholder="Test Placeholder"
    type="text"
    value="test"
  />
</div>
`;
